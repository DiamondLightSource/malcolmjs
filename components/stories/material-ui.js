import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import darkBaseTheme from 'material-ui/styles/baseThemes/darkBaseTheme';
import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import AppBar from 'material-ui/AppBar';
import RaisedButton from 'material-ui/RaisedButton';
import FlatButton from 'material-ui/RaisedButton';
import Toggle from 'material-ui/Toggle';
import { storiesOf, action } from '@kadira/storybook';

function handleEdgeDeleteButton(){

  console.log("material-ui button clicked");

}

storiesOf('Button', module)
    .add('material-ui flat button', () => (
        <MuiThemeProvider  muiTheme={getMuiTheme(lightBaseTheme)}>

            <FlatButton key="edgeDeleteButton" className="pull-right btn btn-default" label="edge Delete Button"
                          onClick={handleEdgeDeleteButton()}>Delete Edge</FlatButton>
        </MuiThemeProvider>
  ))
    .add('material-ui raised button', () => (
        <MuiThemeProvider  muiTheme={getMuiTheme(darkBaseTheme)}>
            <RaisedButton key="edgeDeleteButton" className="pull-right btn btn-default" label="edgeDeleteButton"
                          onClick={action("Raised button clicked")}>Delete Edge</RaisedButton>
        </MuiThemeProvider>
    ))
    .add('material-ui Toggle switch', () => (
        <MuiThemeProvider  muiTheme={getMuiTheme(darkBaseTheme)}>
            <Toggle key="edgeToggleSwitch" className="pull-right btn btn-default" label="Edge Toggle"
                          onToggle={action("Toggle switch")}/>
        </MuiThemeProvider>
    ));

