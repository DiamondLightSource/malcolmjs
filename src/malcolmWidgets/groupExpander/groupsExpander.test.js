import React from 'react';
import { createShallow } from 'material-ui/test-utils';
import GroupExpander from './groupExpander.component';

describe('GroupExpander', () => {
  let shallow;

  beforeEach(() => {
    shallow = createShallow();
  });

  const expander = expanded => (
    <GroupExpander groupName="Test Group" expanded={expanded}>
      <div>Attribute 1</div>
      <div>Attribute 2</div>
      <div>Attribute 3</div>
    </GroupExpander>
  );

  it('renders when default expanded is false', () => {
    const wrapper = shallow(expander(false));
    expect(wrapper).toMatchSnapshot();
  });

  it('renders when default expanded is true', () => {
    const wrapper = shallow(expander(true));
    expect(wrapper).toMatchSnapshot();
  });
});
