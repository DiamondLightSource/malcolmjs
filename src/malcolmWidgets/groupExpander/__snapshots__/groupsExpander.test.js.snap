// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupExpander renders when default expanded is false 1`] = `
<div>
  <WithStyles(ExpansionPanel)
    defaultExpanded={false}
    elevation={0}
  >
    <WithStyles(ExpansionPanelSummary)
      classes={
        Object {
          "content": "WidgetGroupExpander-content-3",
          "expandIcon": "WidgetGroupExpander-expandIcon-4",
          "expanded": "WidgetGroupExpander-expanded-2",
          "root": "WidgetGroupExpander-root-1",
        }
      }
      expandIcon={<pure(ExpandMoreIcon) />}
    >
      <WithStyles(Typography)
        className="WidgetGroupExpander-heading-5"
        variant="subheading"
      >
        Test Group
      </WithStyles(Typography)>
    </WithStyles(ExpansionPanelSummary)>
    <WithStyles(ExpansionPanelDetails)
      classes={
        Object {
          "root": "WidgetGroupExpander-detailsRoot-6",
        }
      }
    >
      <div
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <div>
          Attribute 1
        </div>
        <div>
          Attribute 2
        </div>
        <div>
          Attribute 3
        </div>
      </div>
    </WithStyles(ExpansionPanelDetails)>
  </WithStyles(ExpansionPanel)>
</div>
`;

exports[`GroupExpander renders when default expanded is true 1`] = `
<div>
  <WithStyles(ExpansionPanel)
    defaultExpanded={true}
    elevation={0}
  >
    <WithStyles(ExpansionPanelSummary)
      classes={
        Object {
          "content": "WidgetGroupExpander-content-3",
          "expandIcon": "WidgetGroupExpander-expandIcon-4",
          "expanded": "WidgetGroupExpander-expanded-2",
          "root": "WidgetGroupExpander-root-1",
        }
      }
      expandIcon={<pure(ExpandMoreIcon) />}
    >
      <WithStyles(Typography)
        className="WidgetGroupExpander-heading-5"
        variant="subheading"
      >
        Test Group
      </WithStyles(Typography)>
    </WithStyles(ExpansionPanelSummary)>
    <WithStyles(ExpansionPanelDetails)
      classes={
        Object {
          "root": "WidgetGroupExpander-detailsRoot-6",
        }
      }
    >
      <div
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <div>
          Attribute 1
        </div>
        <div>
          Attribute 2
        </div>
        <div>
          Attribute 3
        </div>
      </div>
    </WithStyles(ExpansionPanelDetails)>
  </WithStyles(ExpansionPanel)>
</div>
`;
